picard/webservice.py

```diff
@@ -187,45 +269,14 @@ def setup_proxy(self):
             proxy.setPassword(config.setting["proxy_password"])
         self.manager.setProxy(proxy)
 
-    def _start_request_continue(self, method, host, port, path, data, handler, xml,
-                                mblogin=False, cacheloadcontrol=None, refresh=None,
-                                access_token=None, queryargs=None):
-        url = build_qurl(host, port, path=path, mblogin=mblogin,
-                         queryargs=queryargs)
-        request = QtNetwork.QNetworkRequest(url)
-        if mblogin and access_token:
-            request.setRawHeader("Authorization", "Bearer %s" % access_token)
-        if mblogin or (method == "GET" and refresh):
-            request.setPriority(QtNetwork.QNetworkRequest.HighPriority)
-            request.setAttribute(QtNetwork.QNetworkRequest.CacheLoadControlAttribute,
-                                 QtNetwork.QNetworkRequest.AlwaysNetwork)
-        elif method == "PUT" or method == "DELETE":
-            request.setPriority(QtNetwork.QNetworkRequest.HighPriority)
-        elif cacheloadcontrol is not None:
-            request.setAttribute(QtNetwork.QNetworkRequest.CacheLoadControlAttribute,
-                                 cacheloadcontrol)
-        request.setRawHeader("User-Agent", USER_AGENT_STRING)
-        if xml:
-            request.setRawHeader("Accept", "application/xml")
-        if data is not None:
-            if method == "POST" and host == config.setting["server_host"] and xml:
-                request.setHeader(QtNetwork.QNetworkRequest.ContentTypeHeader, "application/xml; charset=utf-8")
-            else:
-                request.setHeader(QtNetwork.QNetworkRequest.ContentTypeHeader, "application/x-www-form-urlencoded")
-        send = self._request_methods[method]
-        reply = send(request, data) if data is not None else send(request)
-        self._remember_request_time((host, port))
-        self._active_requests[reply] = (request, handler, xml, refresh)
-
-    def _start_request(self, method, host, port, path, data, handler, xml,
-                       mblogin=False, cacheloadcontrol=None, refresh=None,
-                       queryargs=None):
+    def _start_request(self, wsrequest):
         def start_request_continue(access_token=None):
-            self._start_request_continue(
-                method, host, port, path, data, handler, xml,
-                mblogin=mblogin, cacheloadcontrol=cacheloadcontrol, refresh=refresh,
-                access_token=access_token, queryargs=queryargs)
-        if mblogin and path != "/oauth2/token":
+            wsrequest.access_token = access_token
+            reply = wsrequest.send(self._request_methods)
```

Comment:
Nope, as it depends on manager, it will be a mess to pass it everywheren, i thought about it, and it isn't very clean i agree but for now i prefer to not bother with that.